Zend/zend_compile.c

```diff
@@ -4623,6 +4667,23 @@ void zend_compile_switch(zend_ast *ast) /* {{{ */
 	case_node.op_type = IS_TMP_VAR;
 	case_node.u.op.var = get_temporary_variable(CG(active_op_array));
 
+	jumptable_type = determine_switch_jumptable_type(cases);
+	if (jumptable_type != IS_UNDEF && cases->children >= 4) {
```

Comment:
Here are some benchmarks for the case of 4 children:

```
BEFORE
0_to_3 on 0: 0.13633418083191 s
0_to_3 on 1: 0.16423511505127 s
0_to_3 on 2: 0.19071412086487 s
0_to_3 on 3: 0.23522996902466 s
0_to_3 on 4: 0.19215512275696 s

AFTER
0_to_3 on 0: 0.18555402755737 s
0_to_3 on 1: 0.17891001701355 s
0_to_3 on 2: 0.18369913101196 s
0_to_3 on 3: 0.18191909790039 s
0_to_3 on 4: 0.12473797798157 s

BEFORE
a_to_d on a: 0.16907715797424 s
a_to_d on b: 0.22716999053955 s
a_to_d on c: 0.29214215278625 s
a_to_d on d: 0.35193300247192 s
a_to_d on e: 0.38842010498047 s

AFTER
a_to_d on a: 0.18853902816772 s
a_to_d on b: 0.18467617034912 s
a_to_d on c: 0.1865451335907 s
a_to_d on d: 0.187255859375 s
a_to_d on e: 0.11243486404419 s
```

Maybe it would be better to split integer/string case and use children>=5 for integer and children >=2 for strings.